packages/core-api/src/apis/implementations/StorageApi/WebStorage.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { StorageApi, ObservableMessage } from '../../definitions';
+import { Observable } from '../../../types';
+import ObservableImpl from 'zen-observable';
+
+export class WebStorage implements StorageApi {
+  constructor(private readonly namespace: string = '') {}
+
+  get<T>(key: string): T | undefined {
+    try {
+      const storage = JSON.parse(localStorage.getItem(this.getKeyName(key))!);
+      return storage ?? undefined;
+    } catch (e) {
+      window.console.error(
```

Comment:
I think we could report actual parse errors with the `ErrorApi`? Shouldn't really be a thing that is expected to happen, so good to notify imo.

Reply:
Is there a nice way to grab the `ErrorApi` without hooks? I'm sure there is, just couldn't find much using it already

Reply:
Yup, just add it as a constructor parameter and pass in the ErrorApi instance when it's constructed

Reply:
Don't forget this btw 😁

Reply:
But where does the `new` get called in the frontend app as we only export as a class here. Just still getting to grips with all the plugin architechture

Reply:
Either directly or with automatic DI, check https://github.com/spotify/backstage/blob/master/packages/app/src/apis.ts and https://github.com/spotify/backstage/blob/master/packages/dev-utils/src/devApp/apiFactories.ts. We're only using automatic DI for dev/test envs because it's more complicated but lets you override stuff higher up in the dependency graph.

Reply:
OK - Now I get it - thanks!